src/plugins/intel_cpu/src/nodes/mha.cpp

```diff
@@ -0,0 +1,1086 @@
+
+#define GET_OFF(field) offsetof(jit_mul_add_softmax_call_args, field)
+        mov(reg_mul_in0, ptr[reg_params + GET_OFF(p_mul_in0)]);
+        mov(reg_mul_in1, ptr[reg_params + GET_OFF(p_mul_in1)]);
+        mov(reg_add_in1, ptr[reg_params + GET_OFF(p_add_in1)]);
+        mov(reg_out, ptr[reg_params + GET_OFF(p_out)]);
+        mov(reg_buffer, ptr[reg_params + GET_OFF(p_buffer)]);
+
+        Xbyak::Label mul_add_max_loop_label;
+        Xbyak::Label mul_add_max_end_label;
+        Xbyak::Label sub_exp_reduce_loop_label;
+        Xbyak::Label sub_exp_reduce_end_label;
+        Xbyak::Label mul_loop_label;
+        Xbyak::Label mul_end_label;
+
+        size_t tail_size = jcp_.work_amount % vec_size;
+
+        mov(reg_buffer_aux, reg_buffer);
+        mov(reg_work_amount, jcp_.work_amount);
+        mov(reg_work_amount_aux, reg_work_amount);
+        uni_vpxor(get_vmm_max(0), get_vmm_max(0), get_vmm_max(0));
+
+        // mul1 input is const and always float
+        uni_vmovss(Xmm(get_vmm_in(1).getIdx()), ptr[reg_mul_in1]);
+        uni_vbroadcastss(get_vmm_in(1), Xmm(get_vmm_in(1).getIdx()));
+
+        if (jcp_.with_scales0) {
+            mov(reg_scales, ptr[reg_params + GET_OFF(p_scales0)]);
+
+            mov(reg_tmp, dnnl::impl::float2int(-128.0f));
+            vmovq(xmm_tmp, reg_tmp);
+            vbroadcastss(vmm_crop_low, xmm_tmp);
+
+            mov(reg_tmp, dnnl::impl::float2int(127.0f));
+            vmovq(xmm_tmp, reg_tmp);
+            vbroadcastss(vmm_crop_high, xmm_tmp);
+        }
+
+        if (jcp_.with_scales0 && jcp_.broadcast_scales0) {
+            uni_vmovss(Xmm(vmm_scales.getIdx()), ptr[reg_scales]);
+            uni_vbroadcastss(vmm_scales, Xmm(vmm_scales.getIdx()));
+        }
+
+        L(mul_add_max_loop_label);
+        {
+            cmp(reg_work_amount_aux, vec_size);
+            jl(mul_add_max_end_label, T_NEAR);
+
+            mul_add_max(vec_size);
+
+            sub(reg_work_amount_aux, vec_size);
+
+            jmp(mul_add_max_loop_label, T_NEAR);
+        }
+        L(mul_add_max_end_label);
+        if (tail_size) {
+            mul_add_max(tail_size);
+        }
+
+        sub(rsp, sizeof(float) * vec_size);
+        uni_vmovups(ptr[rsp], get_vmm_max(0));
+        uni_vpxor(get_vmm_max(0), get_vmm_max(0), get_vmm_max(0));
+        for (size_t i = 0; i < vec_size; i++) {
+            mov(reg_tmp_32, ptr[rsp + i * sizeof(float)]);
+            vmovq(xmm_tmp, reg_tmp);
+            uni_vmaxps(get_xmm_max(0), get_xmm_max(0), xmm_tmp);
+        }
+        uni_vbroadcastss(get_vmm_max(0), get_xmm_max(0));
+        add(rsp, sizeof(float) * vec_size);
+
+        uni_vpxor(get_vmm_denom(0), get_vmm_denom(0), get_vmm_denom(0));
+        mov(reg_work_amount_aux, reg_work_amount);
+        mov(reg_buffer_aux, reg_buffer);
+        L(sub_exp_reduce_loop_label);
+        {
+            cmp(reg_work_amount_aux, vec_size);
+            jl(sub_exp_reduce_end_label, T_NEAR);
+
+            sub_exp_reduce(vec_size);
```

Comment:
maybe could be fused in the loop of "mul_add_max(vec_size)" in the future.

Reply:
`mul_add_max` doing reduction to find max value in a roi. We can apply `sub_exp_reduce` only after max is computed.